packages/block-library/src/comments/index.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Server-side rendering fallback of the `core/comments-query-loop` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/comments-query-loop` block fallback on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Returns the filtered post comments for the current post wrapped inside "p" tags.
+ */
+function render_block_core_comments_query_loop( $attributes, $content, $block ) {
+	global $post;
+
+	// Do we have child blocks?
+	if ( count( $block->inner_blocks ) > 0 ) {
+		$inner_blocks_html = '';
+		foreach ( $block->inner_blocks as $inner_block ) {
+			$inner_blocks_html .= $inner_block->render();
+		}
+		return $inner_blocks_html;
+	}
+
+	$post_id = $block->context['postId'];
+	if ( ! isset( $post_id ) ) {
+		return '';
+	}
+
+	$comment_args = array(
+		'post_id' => $post_id,
+		'count'   => true,
+	);
+	// Return early if there are no comments and comments are closed.
+	if ( ! comments_open( $post_id ) && get_comments( $comment_args ) === 0 ) {
+		return '';
+	}
+
+	$post_before = $post;
+	$post        = get_post( $post_id );
+	setup_postdata( $post );
+
+	ob_start();
+	// There's a deprecation warning generated by WP Core.
+	// Ideally this deprecation is removed from Core.
+	// In the meantime, this removes it from the output.
+	add_filter( 'deprecated_file_trigger_error', '__return_false' );
+	comments_template();
+	remove_filter( 'deprecated_file_trigger_error', '__return_false' );
+	$post = $post_before;
+
+	$classes = '';
+	if ( isset( $attributes['textAlign'] ) ) {
+		$classes .= 'has-text-align-' . $attributes['textAlign'];
+	}
+
+	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classes ) );
+	$output             = ob_get_clean();
+
+	wp_enqueue_script( 'comment-reply' );
```

Comment:
Would it not just cleaner to have this as a dependency on the block scripts?

Reply:
You mean as a `viewScript` in `block.json`, like so?

```diff
diff --git a/packages/block-library/src/comments/block.json b/packages/block-library/src/comments/block.json
index 151664b290..57a9aa3772 100644
--- a/packages/block-library/src/comments/block.json
+++ b/packages/block-library/src/comments/block.json
@@ -29,5 +29,6 @@
                }
        },
        "editorStyle": "wp-block-comments-editor",
+       "viewScript": "comment-reply",
        "usesContext": [ "postId", "postType" ]
 }
```

Untested, but if it works -- sounds good to me!

Reply:
(Maybe we can use the same technique also instead of `enqueue_legacy_post_comments_block_styles`?)

Reply:
Makes sense -- let's keep it!